### PR TITLE
DHFPROD-6646: Reverting DHF to Java 8

### PR DESCRIPTION
**Pull request by rjrudin**
### Description

Code changes are mostly changing how @Deprecated is used and not using some collections APIs that were added in Java 9 (but are easily replaced). 

Commented out CreatedOnFacetHandlerTest, will figure out how to fix those lines before this is merged. 

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by ryanjdew — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -18,8 +18,8 @@ allprojects {
 subprojects {
     apply plugin: 'java'
     apply plugin: 'jacoco'
-    sourceCompatibility = "9"
-    targetCompatibility = "9"
+    sourceCompatibility = "8"
+    targetCompatibility = "8"
```
Comment (akshaysonvane) [inline build.gradle:22]: Is there any difference in using "1.8" vs "8"?

Comment (rjrudin) [reply build.gradle:22]: Gradle (actually I think Java) seems good with both - we've been using "8" in the Spark connector for months now.

**Comment by rjrudin** [comment]
Added change from @SameeraPriyathamTadikonda  for Jenkinsfile

File: build.gradle (hunk 1 above)
Comment (akshaysonvane) [reply build.gradle:22]: ahh okay

**Review by akshaysonvane — APPROVED** [review]

**Review by ryanjdew — DISMISSED** [review]

**Comment by rjrudin** [comment]
Fixed tests, as Arrays.asList was breaking - replaced with new ArrayList()

**Review by srinathgit — DISMISSED** [review]

**Review by akshaysonvane — DISMISSED** [review]

**Review by ryanjdew — DISMISSED** [review]

**Comment by rjrudin** [comment]
Added sleep to LoadUserArtifactsCommandTest to prevent intermittent error

**Review by ryanjdew — APPROVED** [review]

**Review by akshaysonvane — APPROVED** [review]

**Pull request merged**